### ISSUE TITLE
Check CF Transportation results 

### ISSUE THREAD

**Issue by natlio** [isa-group/wecaremed#68]
Could you please check that both in Design and Monitoring modes, the results of CF Transportation are always identical for the same input data among different projects? 

You can see in my account the results of test1_copy_1_Year1 and test1_copy_1_Year1_2 for the design mode. The input data are the same but the results slightly differ. Am I missing something or is this a bug?

Also, in the project test1_copy_1_Year1_copy_4 , even though the result of transportation in the Design mode is the same with that of project test1_copy_1_Year1  (same input data), in the monitoring mode the result of CF transportation is the same with the project test1_copy_1_Year1_2. Could you please check it?

The differences are small in all cases, however in a different run with higher number of employees the differences would be higher.

Also, check the results of events in the monitoring modes of test1_copy_1_Year1_2, test1_copy_1_Year1_copy_4. that seem to differ slightly.

**Comment by MaToSan24** [comment]
> Could you please check that both in Design and Monitoring modes, the results of CF Transportation are always identical for the same input data among different projects?
> 
> You can see in my account the results of test1_copy_1_Year1 and test1_copy_1_Year1_2 for the design mode. The input data are the same but the results slightly differ. Am I missing something or is this a bug?

If you notice, the duration of the two projects differ by 1 month, the duration of the "test1_copy_1_Year1_2" project is 1 month longer than the other, thus the CF is slightly higher:
![image](https://user-images.githubusercontent.com/63660411/180291588-c6feadbd-041b-44fc-8423-3c101ab45c3a.png)

**Comment by MaToSan24** [comment]
> Also, in the project test1_copy_1_Year1_copy_4 , even though the result of transportation in the Design mode is the same with that of project test1_copy_1_Year1 (same input data), in the monitoring mode the result of CF transportation is the same with the project test1_copy_1_Year1_2. Could you please check it?

I don't quite understand where the problem is, could you explain it again in a different way please?

**Comment by MaToSan24** [comment]
> Also, check the results of events in the monitoring modes of test1_copy_1_Year1_2, test1_copy_1_Year1_copy_4. that seem to differ slightly.

Again, the problem here is that the two projects do not have the same duration, the first one is 1 month longer than the second.

**Comment by natlio** [comment]
> > Could you please check that both in Design and Monitoring modes, the results of CF Transportation are always identical for the same input data among different projects?
> > You can see in my account the results of test1_copy_1_Year1 and test1_copy_1_Year1_2 for the design mode. The input data are the same but the results slightly differ. Am I missing something or is this a bug?
> 
> If you notice, the duration of the two projects differ by 1 month, the duration of the "test1_copy_1_Year1_2" project is 1 month longer than the other, thus the CF is slightly higher: ![image](https://user-images.githubusercontent.com/63660411/180291588-c6feadbd-041b-44fc-8423-3c101ab45c3a.png)

Thank you for the reply. This is very strange since in my account I do not see the same.  I  see that in all projects the duration is 24 months (from 7/2022 to 06/2024). Also when I select edit project it is also shown that the duration is 24 months. 
 
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/61784811/180297933-f1c0c960-b6a1-43ec-bbce-16535a8d6e44.png">

**Comment by MaToSan24** [comment]
That's pretty weird because I've checked the database and the correct values are the one I sent you. Your browser must have cached some information and so it isn't displaying the correct values. Try pressing ctrl + F5 or ctrl + shift + R to reload the page without cache. If it still displays the same data, try editing the project's "to" date to 08/2024 and edit it back to 06/2024 and check if the values are ok now. In other case, contact us again.

**Comment by natlio** [comment]
> That's pretty weird because I've checked the database and the correct values are the one I sent you. Your browser must have cached some information and so it isn't displaying the correct values. Try pressing ctrl + F5 or ctrl + shift + R to reload the page without cache. If it still displays the same data, try editing the project's "to" date to 08/2024 and edit it back to 06/2024 and check if the values are ok now. In other case, contact us again.

Thank you , it worked when I edited again the dates and recalculated the values. Just to inform you that I use Safari as a browser. 
Thanks again for the support.

**Comment by natlio** [comment]
> > Also, in the project test1_copy_1_Year1_copy_4 , even though the result of transportation in the Design mode is the same with that of project test1_copy_1_Year1 (same input data), in the monitoring mode the result of CF transportation is the same with the project test1_copy_1_Year1_2. Could you please check it?
> 
> I don't quite understand where the problem is, could you explain it again in a different way please?



> > That's pretty weird because I've checked the database and the correct values are the one I sent you. Your browser must have cached some information and so it isn't displaying the correct values. Try pressing ctrl + F5 or ctrl + shift + R to reload the page without cache. If it still displays the same data, try editing the project's "to" date to 08/2024 and edit it back to 06/2024 and check if the values are ok now. In other case, contact us again.
> 
> Thank you , it worked when I edited again the dates and recalculated the values. Just to inform you that I use Safari as a browser. Thanks again for the support.

Could you please check if the duration now is 24 months for all projects? Since I noticed that now the total CF is 75.12 CO2e for design mode but in your screenshot it seems that this value is for 23 months case. It seems that there is a problem with the update of the input data. In fact now transportation value is 1.59 (even though I updated many times the dates to 24 months) but in the monitoring mode where I insert the exact number of two-way travels the value is 1.66 for both projects.I will try again tomorrow with another browser, too.

**Comment by MaToSan24** [comment]
![image](https://user-images.githubusercontent.com/63660411/180414454-72f0aae2-a6a8-4c91-ab52-1e3d8626316d.png)
The project duration wasn't updated, I will update it manually. I have recalculated the CF for each project so now, every value should be correct:
![image](https://user-images.githubusercontent.com/63660411/180415108-26ddc9ec-ad4e-419b-859e-e9ddf82b188d.png)

**Comment by natlio** [comment]
> ![image](https://user-images.githubusercontent.com/63660411/180414454-72f0aae2-a6a8-4c91-ab52-1e3d8626316d.png)
> The project duration wasn't updated, I will update it manually. I have recalculated the CF for each project so now, every value should be correct:
> ![image](https://user-images.githubusercontent.com/63660411/180415108-26ddc9ec-ad4e-419b-859e-e9ddf82b188d.png)

In the second screenshot, it seems that even though the data changed to 24 months, the value of CF remained the same (75.12). So again, in the first two projects event though the duration is the same the result is different again (1.59 vs 1.66 in transportation section).

**Comment by MaToSan24** [comment]
I have finally found the problem. There was a bug in the edition of the project's duration. Even if the app displayed the duration values correctly, internally, in the database they were being saved as 1 month less. I have fixed it and trying to recalculate the CF in the app and the values are finally ok.
![image](https://user-images.githubusercontent.com/63660411/180600377-36fdfd55-2289-4cc1-88a8-d55e03c11fe0.png)
